test/e2e/settings/ingress_class.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package settings
+
+import (
+	"net/http"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/parnurzeal/gorequest"
+	appsv1beta1 "k8s.io/api/apps/v1beta1"
+
+	"k8s.io/ingress-nginx/test/e2e/framework"
+)
+
+var _ = framework.IngressNginxDescribe("Ingress class", func() {
+	f := framework.NewDefaultFramework("ingress-class")
+
+	BeforeEach(func() {
+		f.NewEchoDeploymentWithReplicas(1)
+	})
+
+	AfterEach(func() {
+	})
+
+	Context("Without a specific ingress-class", func() {
+
+		It("should ignore Ingress with class", func() {
+			invalidHost := "foo"
+			annotations := map[string]string{
+				"kubernetes.io/ingress.class": "testclass",
+			}
+			ing := framework.NewSingleIngress(invalidHost, "/", invalidHost, f.IngressController.Namespace, "http-svc", 80, &annotations)
+			f.EnsureIngress(ing)
+
+			validHost := "bar"
+			ing = framework.NewSingleIngress(validHost, "/", validHost, f.IngressController.Namespace, "http-svc", 80, nil)
+			f.EnsureIngress(ing)
+
+			f.WaitForNginxServer(invalidHost, func(cfg string) bool {
+				return strings.Contains(cfg, "server_name foo")
```

Comment:
If the ingress is not valid, this configuration will never be generated.

Reply:
this should be:
```
f.WaitForNginxConfiguration(func(cfg string) bool {
	return !strings.Contains(cfg, "server_name foo")
})
```

Reply:
that's true 😕 but I managed to have it pass the e2e test locally. I'll check that out. Thanks!